test/support/CMakeLists.txt

```diff
@@ -0,0 +1,108 @@
+#
+# test/CMakeLists.txt
+#
+#
+# The MIT License
+#
+# Copyright (c) 2022 TileDB, Inc.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+find_package(Catch_EP REQUIRED)
+
+# Include TileDB core header directories
+set(TILEDB_CORE_INCLUDE_DIR "${CMAKE_SOURCE_DIR}")
+
+# Include the C API directory so that the C++ 'tiledb' file can directly
+# include "tiledb.h".
+list(APPEND TILEDB_CORE_INCLUDE_DIR "${CMAKE_SOURCE_DIR}/tiledb/sm/c_api")
+
+# Gather the test source files
+set(TILEDB_TEST_SUPPORT_SOURCES
+  src/helpers.h
+  src/helpers.cc
+  src/helpers-dimension.h
+  src/serialization_wrappers.cc
+  src/vfs_helpers.cc
+  )
+
+if (TILEDB_VERBOSE)
+  add_definitions(-DTILEDB_VERBOSE)
+endif()
+
+add_library(tiledb_test_support_lib EXCLUDE_FROM_ALL
+  $<TARGET_OBJECTS:TILEDB_CORE_OBJECTS>

```

Comment:
Because _ILIB does not provide the sources/objects expected/required by the add_library() call, with those sources/objects part of TILEDB_CORE_OBJECTS providing effective static linkage to anything from core that the support library itself may depend on, and thus available for anything that might use the support library outside of the original/base tiledb_unit executable.
Note that the same reference ($<TARGET_OBJECTS:TILEDB_CORE_OBJECTS>) is used in constructing the tiledb_unit executable,